tests/testthat/test-write.R

```diff
@@ -0,0 +1,174 @@
+redcap_data_a <- tibble::tribble(
+  ~record_id, ~col_a,
+  1,          "A"
+)
+
+redcap_metadata_a <- tibble::tribble(
+  ~field_name, ~field_label,
+  "record_id",  "Record ID",
+  "col_a",      "Label A"
+)
+
+redcap_data_b <- tibble::tribble(
+  ~record_id, ~col_b,
+  1,          "B"
+)
+
+redcap_metadata_b <- tibble::tribble(
+  ~field_name, ~field_label,
+  "record_id",  "Record ID",
+  "col_b",      "Label B"
+)
+
+supertbl <- tibble::tribble(
+  ~redcap_form_name, ~redcap_form_label, ~redcap_data,   ~redcap_metadata,
+  "a",                "A",                redcap_data_a,  redcap_metadata_a,
+  "b",                "B",                redcap_data_b,  redcap_metadata_b
+) %>%
+  as_supertbl()
+
+test_that("write_redcap_xlsx without labels works", {
+  withr::with_dir(
+    tempdir(), {
+      write_redcap_xlsx(supertbl,
+                        file = paste0(tempdir(), "supertbl_wb.xlsx"),
+                        include_metadata = FALSE,
+                        include_toc_from_supertbl = FALSE)
+      sheet_1 <- openxlsx2::read_xlsx(xlsxFile = paste0(tempdir(), "supertbl_wb.xlsx"), sheet = 1, startRow = 1)
+      # For some reason, read_xlsx resets row names and starts at 2, likely due
+      # to reading the column names as a row
+      rownames(sheet_1) <- seq_len(nrow(sheet_1))
+
+      sheet_2 <- openxlsx2::read_xlsx(xlsxFile = paste0(tempdir(), "supertbl_wb.xlsx"), sheet = 2)
+      rownames(sheet_2) <- seq_len(nrow(sheet_2))
+
+      # Ignore attr applied by openxlsx2 read_xlsx
+      expect_equal(tibble::tibble(sheet_1), redcap_data_a, ignore_attr = TRUE)
+      expect_equal(tibble::tibble(sheet_2), redcap_data_b, ignore_attr = TRUE)
+
+      unlink(paste0(tempdir(), "supertbl_wb.xlsx"))
+    }
+  )
+})
+
+
+test_that("write_redcap_xlsx with labels works", {
+  labelled_supertbl <- make_labelled(supertbl)
+
+  labelled_sheet_1 <- tibble::tribble(
+    ~"Record.ID", ~"Label.A",
+    "record_id",   "col_a",
+    "1",           "A"
+  )
+
+  labelled_sheet_2 <- tibble::tribble(
+    ~"Record.ID", ~"Label.B",
+    "record_id",   "col_b",
+    "1",           "B"
+  )
+
+  withr::with_dir(
```

Comment:
Consider `with_tempdir()` so you don't have to `unlink()`

Reply:
Ah good call, total oversight on my part